File: CPP/DataModel/ChatMsg.cpp
#include "ChatMsg.h"

ChatMsg::ChatMsg()
{
	_type = 0;
}

void ChatMsg::Load(QJsonObject obj)
{
	if (!obj.isEmpty())
	{
		JsonUtil::GetValue(obj, "account_id", _account_id);
		JsonUtil::GetValue(obj, "room_id", _room_id);
		JsonUtil::GetValue(obj, "account_name", _account_name);
		JsonUtil::GetValue(obj, "account_portrait", _account_portrait);
		JsonUtil::GetValue(obj, "target_id", _target_id);
		JsonUtil::GetValue(obj, "message", _message);
		JsonUtil::GetValue(obj, "type", _type);
	}
}

File: CSharp/vcsSdk/Models/ChatMsg.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace vcsSdk.Models
{
    public class ChatMsg
    {
        [JsonProperty("account_id")]
        public string AccountId { get; set; }
        [JsonProperty("room_id")]
        public string RoomId { get; set; }
        [JsonProperty("account_name")]
        public string AccountName { get; set; }
        [JsonProperty("account_portrait")]
        public string AccountPortrait { get; set; }
        [JsonProperty("target_id")]
        public string TargetId { get; set; }
        [JsonProperty("message")]
        public string Message { get; set; }
        [JsonProperty("type")]
        public int Type { get; set; }
    }
}
